core/logging.h

```diff
@@ -39,19 +40,28 @@ static inline void log(
     userMessage += " " + extra;
   }
 
-  std::string message = Port::getHumanTime();
+  // TODO: this is not a cleanest solution, but it works for now
+
+  std::string message;
+
+#ifndef IDF_TARGET
+  message = Port::getHumanTime();
   message += " " + padRight(8, level._to_string());
+#endif
 
-#ifdef DEBUG_BUILD
+#if defined(DEBUG_BUILD) && not defined(IDF_TARGET)
   message += " " + padRight(80, userMessage);
   message +=
       " (" + stripLogPathPrefix(filename) + ":" + std::to_string(lineno) + ")";
 #else
   message += " " + userMessage;
 #endif
 
-  Port::log(message);
+  Port::log(level, message);
+
+#ifndef IDF_TARGET
```

Comment:
This however kinda feels like should be brought back. I'm not sure whether log manager is absolutely necessary on this platform but… on the other hand it seems like the best way of getting logs in a non-intrusive way

Reply:
ESP32 does not use`LogManager` for any functionality. Current implementation rapidly allocates new strings and stores N last entries. This results in a heavy memory fragmentation. I would drop it in ESP32 port for now. In future if this feature would be required it should utilize some sort of a static message buffer.